src_c/camera_windows.c

```diff
@@ -0,0 +1,862 @@
+/*
+  pygame - Python Game Library
+  Copyright (C) 2000-2001  Pete Shinners
+
+  This library is free software; you can redistribute it and/or
+  modify it under the terms of the GNU Library General Public
+  License as published by the Free Software Foundation; either
+  version 2 of the License, or (at your option) any later version.
+
+  This library is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+  Library General Public License for more details.
+
+  You should have received a copy of the GNU Library General Public
+  License along with this library; if not, write to the Free
+  Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+
+  Pete Shinners
+  pete@shinners.org
+*/
+
+/*
+ * Windows Camera - webcam support for pygame
+ * Original Author: Charlie Hayden, 2021
+ * 
+ * This sub-module adds native support for windows webcams to pygame,
+ * made possible by Microsoft Media Foundation.
+ * 
+ * Since Media Foundation is still maturing, support is best in Windows 10 or
+ * above, but I believe this will work in Windows 8 as well.
+ * 
+ * The Media Foundation documentation is written for use in C++, but the API
+ * supports C as well. What would be a call to a class method in C++ is a call
+ * through a pointer table lpVtbl.
+ * For example obj->func(arguments) would be obj->lpVtbl->func(obj, arguments)
+ * Also, GUIDs need to be addressed using '&' to work.
```

Comment:
Those documentation URLs would be helpful here.

Reply:
This was something I found out on stackoverflow :cry:
I don't know if it's documented. It certainly wasn't in **any** of the media foundation pages I read

Reply:
I just meant, adding those two URLs and your note about the threading architecture in here would be helpful.